Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -100,9 +141,36 @@ public override void Close()
             _connectionState = ConnectionState.Closed;
         }
 
+        public override void Close()
+        {
+            SnowflakeDbConnection conn = SnowflakeDbConnectionPool.getConnection(this.ConnectionString, this.Password);
+            if (conn != null)
+            {
+                conn._refCount--;

```

Comment:
I am afraid the update of _refCount needs to be protected, like Interlocked.increment/decrement?